File: 0101_0200/0114/0114.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, k int
	fmt.Fscan(f, &n, &k)
	r := (k - 1) * (k - 1)
	m := k - 1
	for i := 2; i < n; i++ {
		m, r = r, (r+m)*(k-1)
	}
	fmt.Fprint(q, r+m)
}

File: 0101_0200/0114/0114.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, k = map(int, f.read().split())
    r, m = (k - 1) * (k - 1), k - 1
    for i in range(n - 2):
        m, r = r, (r + m) * (k - 1)
    q.write(str(r + m))
